controllers/firewall/status.go

```diff
@@ -106,5 +113,60 @@ func getFirewallNetworks(ctx context.Context, cache *cache.Cache[string, *models
 		})
 	}
 
-	return result, nil
+	r.Target.Status.FirewallNetworks = result
+
+	return nil
+}
+
+func SetFirewallStatus(fw *v2.Firewall, mon *v2.FirewallMonitor) {
+	if v2.IsAnnotationTrue(fw, v2.FirewallNoControllerConnectionAnnotation) {
+		cond := v2.NewCondition(v2.FirewallControllerConnected, v2.ConditionTrue, "NotChecking", "Not checking controller connection due to firewall annotation.")
+		fw.Status.Conditions.Set(cond)
+
+		cond = v2.NewCondition(v2.FirewallDistanceConfigured, v2.ConditionTrue, "NotChecking", "Not checking distance due to firewall annotation.")
+		fw.Status.Conditions.Set(cond)
+
+		return
+	}
+
+	if mon == nil {
+		return
+	}
+
+	if mon.ControllerStatus == nil {
+		cond := v2.NewCondition(v2.FirewallControllerConnected, v2.ConditionFalse, "NotConnected", "Controller has not yet connected.")
+		fw.Status.Conditions.Set(cond)
+
+		cond = v2.NewCondition(v2.FirewallDistanceConfigured, v2.ConditionFalse, "NotConnected", "Controller has not yet connected.")
+		fw.Status.Conditions.Set(cond)
+
+		return
+	}
+
+	connection := &v2.ControllerConnection{
+		ActualVersion:  mon.ControllerStatus.ControllerVersion,
+		Updated:        mon.ControllerStatus.Updated,
+		ActualDistance: mon.ControllerStatus.Distance,
+	}
+
+	fw.Status.ControllerStatus = connection
+
+	if connection.Updated.Time.IsZero() {
+		cond := v2.NewCondition(v2.FirewallControllerConnected, v2.ConditionFalse, "NotConnected", "Controller has not yet connected.")
+		fw.Status.Conditions.Set(cond)
+	} else if time.Since(connection.Updated.Time) > 5*time.Minute {
+		cond := v2.NewCondition(v2.FirewallControllerConnected, v2.ConditionFalse, "StoppedReconciling", fmt.Sprintf("Controller has stopped reconciling since %s.", connection.Updated.Time.String()))
+		fw.Status.Conditions.Set(cond)
+	} else {
+		cond := v2.NewCondition(v2.FirewallControllerConnected, v2.ConditionTrue, "Connected", fmt.Sprintf("Controller reconciled firewall at %s.", connection.Updated.Time.String()))
+		fw.Status.Conditions.Set(cond)
+	}
+
+	if fw.Distance == connection.ActualDistance {
```

Comment:
We probably need to skip this check for firewall-controller versions that do not support this feature. Otherwise the firewall set does not become ready.

Reply:
Added a field for the firewall-controller to tell if it supports distances.